docs/module-caching.md

```diff
@@ -0,0 +1,181 @@
+   assert(c2 === c2);
+   
+   // file.js
+   export const check = {};
+   ```
+3. Importing the same module block twice _from the same Realm_, even if from different modules, results in a single evaluation:
+   ```js
+   globalThis.count = 0;
+   const mod = module { globalThis.count++; };
+   await import(mod);
+   await import(mod);
+   assert(globalThis.count === 1);
+   ```
+   ```js
+   // main.js
+   import { mod, check as c1 } from "./dep.js";
+   const { check: c2 } = await import(mod);
+   assert(c1 === c2);
+
+   // dep.js
+   export const mod = module { export const check = {}; };
+   export const { check } = await import(mod);
+   ```
+4. A module block is evaluated in the same realm as where it is imported:
+   ```js
+   const mod = module { globalThis.modEvaluated = true; };
+   const realm = createLegacyRealm(); // [1]
+   await realm.eval(`s => import(s)`)(mod);
+   assert(globalThis.modEvaluated === undefined);
+   assert(realm.modEvaluated === true);
+   ```
+   This is consistent with the behavior of imports with string specifiers.
+5. Importing the same module block twice _from different Realms_ results in multiple evaluations:
+   ```js
+   const realm = createLegacyRealm(); // [1]
+   globalThis.count = 0;
+   realm.count = 0;
+   const mod = module { globalThis.count++; export const check = {}; };
+   const { check: c1 } = await import(mod);
+   const { check: c2 } = await realm.eval("s => import(s)")(mod);
+   assert(globalThis.count + realm.count === 2);
+   assert(c1 !== c2);
+   ```
+   This is consistent with the behavior of imports with string specifiers.
+6. Importing two different module blocks results in two evaluations and two different namespace objects. Two module blocks `mod1` and `mod2` are considered different if `mod1 !== mod2`:
+   ```js
+   globalThis.count = 0;
+   const mod1 = module { globalThis.count++; export const check = {}; };
+   const mod2 = module { globalThis.count++; export const check = {}; };
+   const { check: c1 } = await import(mod1);
+   const { check: c1 } = await import(mod2);
+   assert(c1 !== c2);
+   assert(globalThis.count === 2);
+   ```
+   ```js
+   globalThis.count = 0;
+   const mod = module { globalThis.count++; export const check = {} };
+   const modClone = structuredClone(mod);
+   assert(mod !== modClone);
+   const { check: c1 } = await import(mod);
+   const { check: c2 } = await import(modClone);
+   assert(c1 !== c2);
+   assert(globalThis.count === 2);
+   ```
+   ```js
+   globalThis.count = 0;
+   const mod = module {};
+   const worker = new Worker(`data:text/javascript,
+     addEventListener("message", msg => postMessage(msg.data))
+   `);
+   worker.addEventListener("message", ({ data: mod2 }) => {
+     assert(mod !== mod2);
+     const { check: c1 } = await import(mod);
+     const { check: c2 } = await import(mod2);
+     assert(c1 !== c2);
+     assert(globalThis.count === 2);
+   });
+   worker.postMessage(mod);
+   ```
```

Comment:
Does this example add value when we have the `structuredClone`?

Reply:
Nope, I can remove it. The worker example is only important in (7) because it shows that resolution should be preserved even across agents.